benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/CustomHashTableJumpTable.cs

```diff
@@ -0,0 +1,191 @@
+                if (!map.TryGetValue(key, out var matches))
+                {
+                    matches = new List<(string text, int destination)>();
+                    map.Add(key, matches);
+                }
+
+                matches.Add(entries[i]);
+            }
+
+            _prime = GetPrime(map.Count);
+            _buckets = new int[_prime + 1];
+            _entries = new Entry[map.Sum(kvp => kvp.Value.Count)];
+
+            var next = 0;
+            foreach (var group in map.GroupBy(kvp => kvp.Key % _prime).OrderBy(g => g.Key))
+            {
+                _buckets[group.Key] = next;
+
+                foreach (var array in group)
+                {
+                    for (var i = 0; i < array.Value.Count; i++)
+                    {
+                        _entries[next++] = new Entry(array.Value[i].text, array.Value[i].destination);
+                    }
+                }
+            }
+
+            Debug.Assert(next == _entries.Length);
+            _buckets[_prime] = next;
+
+            var last = 0;
+            for (var i = 0; i < _buckets.Length; i++)
+            {
+                if (_buckets[i] == 0)
+                {
+                    _buckets[i] = last;
+                }
+                else
+                {
+                    last = _buckets[i];
+                }
+            }
+        }
+
+        public int Find(int key)
+        {
+            return key % _prime;
+        }
+        private static int GetPrime(int capacity)
+        {
+            for (int i = 0; i < Primes.Length; i++)
+            {
+                int prime = Primes[i];
+                if (prime >= capacity)
+                {
+                    return prime;
+                }
+            }
+
+            return Primes[Primes.Length - 1];
+        }
+
+        public override unsafe int GetDestination(string path, PathSegment segment)
+        {
+            if (segment.Length == 0)
+            {
+                return _exitDestination;
+            }
+
+            var key = GetKey(path, segment);
+            var index = Find(key);
+
+            var start = _buckets[index];
+            var end = _buckets[index + 1];
+
+            var entries = _entries.AsSpan(start, end - start);
+            for (var i = 0; i < entries.Length; i++)
+            {
+                var text = entries[i].Text;
+                if (text.Length == segment.Length &&
+                    string.Compare(
+                        path,
+                        segment.Start,
+                        text,
+                        0,
+                        segment.Length,
+                        StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    return entries[i].Destination;
+                }
+            }
+
+            return _defaultDestination;
+        }
+
+        private static unsafe int GetKey(string path, PathSegment segment)
+        {
+            fixed (char* p = path)
```

Comment:
Comment what you're doing in GetKey? I'm ok leaving it out since this is test code at the moment